packages/server/graphql/types/AzureDevopsAuth.ts

```diff
@@ -0,0 +1,61 @@
+import {GraphQLBoolean, GraphQLID, GraphQLList, GraphQLNonNull, GraphQLObjectType} from 'graphql'
+import GraphQLISO8601Type from './GraphQLISO8601Type'
+import {getUserId} from '../../utils/authorization'
+
+const AzureDevopsAuth = new GraphQLObjectType({
+  name: 'AzureDevopsAuth',
+  description: 'OAuth token for a team member',
+  fields: () => ({
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'shortid'
+    },
+    isActive: {
+      description: 'true if the auth is valid, else false',
+      type: new GraphQLNonNull(GraphQLBoolean)
+    },
+    accessToken: {
+      description:
+        'The access token to Azure Devops, useful for 1 hour. null if no access token available',
+      type: GraphQLID,
+      resolve: async ({teamId, userId}, _args, {authToken, dataLoader}) => {
+        const viewerId = getUserId(authToken)
+        if (viewerId !== userId) return null
+        return dataLoader.get('freshAzureDevopsAccessToken').load({teamId, userId: viewerId})
```

Comment:
there will likely be some nuance here. Jira provides a JWT with a 30 minute expiration. If devops gives a longer timewindow we could probably ignore the refresh entirely